src/main.cpp

```diff
@@ -302,11 +313,28 @@ void nimble_port_ll_task_func(void* args) {
 }
 }
 
+void calibrate_lf_clock_rc(nrf_drv_clock_evt_type_t event) {
+  // 16 * 0.25s = 4s calibration cycle
+  // Not recursive, call is deferred via internal calibration timer
+  nrf_drv_clock_calibration_start(16, calibrate_lf_clock_rc);
+}
+
 int main(void) {
   logger.Init();
 
   nrf_drv_clock_init();
 
+  // In some cases (eg. when a watchdog is enabled), 
+  // the NRF SDK incorrectly assumes the LF clock is already running,
+  nrf_drv_clock_lfclk_request(NULL);
+  nrfx_clock_lfclk_start();
+  while(!nrf_clock_lf_is_running()) { }
```

Comment:
Is this actually a potential issue on the PineTime, should it be wrapped in a preprocessor if?

Reply:
These three lines ensure that the clock is running, which is required by the call `nrf_drv_clock_calibration_start` below. 

On the PineTime, no LFRC calibration is used, and the LF clock is started at a later point at `vPortSetupTimerInterrupt` in `FreeRTOS/port_cmsis_systick.c`. The call to `nrf_drv_clock_lfclk_request` is idempotent.

Do you think these three lines should be moved into the `#if(CLOCK_CONFIG_LF_SRC == NRF_CLOCK_LFCLK_RC)` conditional?

Reply:
This discussion can be moved to https://github.com/InfiniTimeOrg/InfiniTime/pull/1128 , where this change was broken out into.